guides/source/action_view_overview.md

```diff
@@ -318,6 +318,24 @@ You can also specify a second partial to be rendered between instances of the ma
 
 Rails will render the `_product_ruler` partial (with no data passed to it) between each pair of `_product` partials.
 
+#### Strict Templates
+
+By default, templates will accept any `locals`. To define what locals a template should accept, use the `strict` magic comment:
+
+```erb
+<%# strict: (message:) -%>
+<%= message %>
+```
+
+Default values can also be provided:
+
+```erb
+<%# strict: (message: "Hello, world!") -%>
+<%= message %>
+```
+
+To disallow locals entirely, set `# strict: (true)`.
```

Comment:
Floating an idea: what if the magic "comment" was 100% Ruby?  e.g.:

```erb
<% template_signature { |foo:, bar: 2, baz: foo + bar| } %>
```

That might make the valid syntax of the "comment" more obvious, including when disallowing locals:


```erb
<% template_signature { || } %>
```

And it could leverage syntax highlighting in editors.

Reply:
Huh that's pretty cool! I love the idea of having it not be a comment. Are you thinking we would we AST this line to extract the method signature for the template?